src/Compilers/Core/Portable/RuleSet/RuleSetInclude.cs

```diff
@@ -77,42 +77,53 @@ public RuleSet LoadRuleSet(RuleSet parent)
         /// <param name="parent">The parent of this rule set include</param>
         private string GetIncludePath(RuleSet parent)
         {
-            List<string> found = new List<string>();
-            string expandedPath = PortableShim.Environment.ExpandEnvironmentVariables(_includePath);
+            string expandedIncludePath = PortableShim.Environment.ExpandEnvironmentVariables(_includePath);
+            var parentRulesetPath = parent?.FilePath;
 
-            if (PathUtilities.IsUnixLikePlatform)
+            var resolvedIncludePath = ResolveIncludePath(expandedIncludePath, parentRulesetPath);
+            if (resolvedIncludePath == null && PathUtilities.IsUnixLikePlatform)
             {
-                expandedPath = expandedPath.Replace('\\', PortableShim.Path.DirectorySeparatorChar);
+                // Attempt to resolve legacy ruleset includes by replacing Windows style directory separator char with current plaform's directory separator char.
+                resolvedIncludePath = ResolveIncludePath(expandedIncludePath, parentRulesetPath, replaceDirectorySeparatorChar: true);
             }
 
-            // If a full path is specified then use it
-            if (Path.IsPathRooted(expandedPath))
+            // If we still couldn't find it then throw an exception;
+            if (resolvedIncludePath == null)
             {
-                if (PortableShim.File.Exists(expandedPath))
-                {
-                    found.Add(expandedPath);
-                }
+                throw new FileNotFoundException(string.Format(CodeAnalysisResources.FailedToResolveRuleSetName, _includePath), _includePath);
             }
 
-            // If the current rule set is backed by a file then try to find the include file relative to it
-            if (parent != null && !string.IsNullOrEmpty(parent.FilePath))
+            // Return the canonical full path
+            return PortableShim.Path.GetFullPath(resolvedIncludePath);
+        }
+
+        private static string ResolveIncludePath(string includePath, string parentRulesetPath, bool replaceDirectorySeparatorChar = false)
```

Comment:
It seems odd to pass this as an optional parameter. I'd just move the replacement to the caller.